pkg/api/dashboard_permission.go

```diff
@@ -203,12 +203,13 @@ func (hs *HTTPServer) updateDashboardAccessControl(ctx context.Context, orgID in
 		}
 	}
 
-	svc := hs.permissionServices.GetDashboardService()
 	if isFolder {
-		svc = hs.permissionServices.GetFolderService()
+		if _, err := hs.folderPermissionsService.SetPermissions(ctx, orgID, uid, commands...); err != nil {
```

Comment:
Shouldn't you return all the time to prevent the call to `dashboardPermissionsService.SetPermissions` when dealing with a folder?

Reply:
No we call this one from both dashboard and folder permissions endpoints